components/scream/scripts/utils.py

```diff
@@ -298,39 +298,71 @@ def get_cpu_core_count():
     return int(run_cmd_no_fail("cat /proc/cpuinfo | grep processor | wc -l"))
 
 ###############################################################################
-def _ensure_pylib_impl(libname, pipname=None):
+def ensure_pip():
+###############################################################################
+    """
+    Ensures that pip is available. Notice that we cannot use the _ensure_pylib_impl
+    function below, since it would cause circular dependencies. This one has to
+    be done by hand.
+    """
+    # Use ensurepip for installing pip
+    import ensurepip
+    ensurepip.bootstrap(user=True)
+
+    # needed to "rehash" available libs
+    site.main() # pylint: disable=no-member
+
+    _ = import_module("pip")
+
+###############################################################################
+def pip_install_lib(min_version=None, pipname=None):
```

Comment:
I'm wondering if the way min_version is used by this function is what we want. Pip should give you the most recent version that it can, so I don't think we should be asking for the minimum. One thing we could do is verify that the version pip gave us is above the minimum like you did in the function below.

Reply:
I guess a different way could be to do `pip install --upgrade $pipname` if version is present but not at the right version.

Reply:
Yeah, that would be a nice addition.